benches/seq_choose.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright 2018-2022 Developers of the Rand project.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+use criterion::{criterion_group, criterion_main, Criterion, black_box};
+use rand::prelude::*;
+
+use rand_chacha::ChaCha20Rng as CryptoRng;
+// We force use of 32-bit RNG since seq code is optimised for use with 32-bit
+// generators on all platforms.
+use rand_pcg::Pcg32 as SmallRng;
```

Comment:
It is... so we should ignore results with other RNGs? The current `SmallRng` on 64-bit platforms is `Xoshiro256PlusPlus` (64-bit native), so we should at least check what performance with this RNG looks like (this probably means using three RNGs: ChaCha as a block RNG and either PCG or Xoshiro in both 32- and 64-bit variants).

Reply:
Good idea. I've reworked the benchmarks to make changing the RNG easier and added PCG64. I also tested with Xoshiro256PlusPlus.  The performance changes for both look pretty similar to the 32 bit rngs - 30-50% speedups for the unhinted and stable cases, very small regressions for the others.